packages/langchain/src/openai/util.test.ts

```diff
@@ -2,55 +2,71 @@ import {
   AzureOpenAiCreateChatCompletionResponse,
   AzureOpenAiCreateChatCompletionRequest
 } from '@sap-ai-sdk/foundation-models';
-import nock from 'nock';
-import { BaseMessage, HumanMessage } from '@langchain/core/messages';
 import {
-  mockClientCredentialsGrantCall,
-  parseMockResponse
-} from '../../../../test-util/mock-http.js';
+  BaseMessage,
+  RemoveMessage,
+  ToolMessage
+} from '@langchain/core/messages';
+import { parseMockResponse } from '../../../../test-util/mock-http.js';
 import { mapLangchainToAiClient, mapOutputToChatResult } from './util.js';
 import { AzureOpenAiChatClient } from './chat.js';
+import { AzureOpenAiChatCallOptions } from './types.js';
 
 const openAiMockResponse =
   parseMockResponse<AzureOpenAiCreateChatCompletionResponse>(
     'foundation-models',
     'azure-openai-chat-completion-success-response.json'
   );
 
-describe('Mapping Functions', () => {
-  beforeEach(() => {
-    mockClientCredentialsGrantCall();
-  });
-
-  afterEach(() => {
-    nock.cleanAll();
-  });
+// Signal and Prompt Index are provided by the super class in every call
+const defaultOptions = {
+  signal: undefined,
+  promptIndex: 0
+};
 
+describe('Mapping Functions', () => {
   it('should parse an OpenAI response to a (LangChain) chat response', async () => {
     const result = mapOutputToChatResult(openAiMockResponse);
     expect(result).toMatchSnapshot();
   });
 
   it('should parse a LangChain input to an AI SDK input', async () => {
     const langchainPrompt: BaseMessage[] = [
-      new HumanMessage('Where is the deepest place on earth located')
+      new ToolMessage('Test Content', 'test-id')
     ];
 
     const request: AzureOpenAiCreateChatCompletionRequest = {
       messages: [
         {
-          role: 'user',
-          content: 'Where is the deepest place on earth located'
+          role: 'tool',
+          content: 'Test Content',
+          tool_call_id: 'test-id'
         }
-      ]
+      ],
+      tools: [{ type: 'function', function: { name: 'test', parameters: {} } }],
+      tool_choice: 'auto',
+      functions: [{ name: 'random' }, { name: 'test' }]
     };
     const client = new AzureOpenAiChatClient({ modelName: 'gpt-35-turbo' });
-    const defaultOptions = { signal: undefined, promptIndex: 0 };
-    const mapping = mapLangchainToAiClient(
-      client,
-      langchainPrompt,
-      defaultOptions
-    );
+    const options: AzureOpenAiChatCallOptions & {
+      promptIndex?: number;
+    } = {
```

Comment:
Would end users have to use the same approach?

Reply:
Not exactly, users can't even provide signal & promptIndex I believe, however, these values are always part of the options as these super class provides them.